### PR TITLE
added docs for data collected in iOS

### PR DESCRIPTION
**Pull request by priyanshi-gupta**
### What and why?

Adds docs for metrics and attributes collected by iOS SDK.

### How?

Referenced from in-code, view schemas

### Review checklist

- [ ] Make sure all the attributes are covered
- [ ] Make sure each attribute description makes sense in mobile (iOS) context

### CODE REVIEW

File: docs/mobile_data_collected.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+| Event Type | Description                                                                                                                                                                                                                                                   |
+|------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Session    | Session represents a real user journey on your mobile application. It begins when the user launches the application and it is kept live as long as the user stays active. During the user journey, all RUM events generated as part of the session will share the same `session.id` attribute.
+ |
+| View       | View represents a unique screen (or screen segment) on your mobile application. Individual ViewControllers are classified as distinct views. While a user stays on a view, RUM event attributes (Errors, Resources, Actions) get attached to the View with a unique `view.id`                                   |
```
Comment (apigirl) [inline docs/mobile_data_collected.md:9]:
```suggestion
| View       | View represents a unique screen (or screen segment) on your mobile application. Individual `ViewControllers` are classified as distinct views. While a user stays on a view, RUM event attributes (Errors, Resources, Actions) get attached to the view with a unique `view.id`                                   |
```

File: docs/mobile_data_collected.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+| Resource   | Resource represents network requests to first-party hosts, APIs, 3rd party providers and libraries in your mobile application. All requests generated during a user session are attached to the view with a unique `resource.id`                                                      |
```
Comment (apigirl) [inline docs/mobile_data_collected.md:10]:
```suggestion
| Resource   | Resources represents network requests to first-party hosts, APIs, 3rd party providers, and libraries in your mobile application. All requests generated during a user session are attached to the view with a unique `resource.id`                                                      |
```

File: docs/mobile_data_collected.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+Metrics are quantifiable values that can be used for measurements related to the event. Attributes are non-quantifiable values used to slice metrics data (group by) in analytics. 
+
+{{% /tab %}}
+{{% tab "View" %}}
+
+### View Metrics
+
+
+| Metric                              | Type        | Decription                                                                                          |
```
Comment (apigirl) [inline docs/mobile_data_collected.md:62]:
```suggestion
| Metric                              | Type        | Description                                                                                          |
```

File: docs/mobile_data_collected.md [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+The following diagram illustrates the RUM event hierarchy:
+
+{{< img src="real_user_monitoring/data_collected/event-hierarchy.png" alt="RUM Event hierarchy" style="width:50%;border:none" >}}
+
+## Default attributes
+
+RUM collects common attributes for all events and attributes specific to each event by default listed below. You can also choose to enrich your user session data with [additional events][1] to default events specific to your application monitoring and business analytics needs.
+
+
+### Common Core Attributes
```
Comment (apigirl) [inline docs/mobile_data_collected.md:23]:
```suggestion
### Common core attributes
```

File: docs/mobile_data_collected.md [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+| `view.name` | string | Name of the initial view corresponding to the event. |
+| `service` | string | The [unified service name][] for this application used to corelate user sessions. |
+| `date` | integer  | Start of the event in ms from epoch. |
+| `session.type` | string | Type of the session (`user`). |
+| `connectivity.status` | string | Status of device connectivity (`connected`, `not connected`, `maybe`). |
+| `connectivity.interfaces` | string | The list of available network interfaces (for example, `bluetooth`, `cellular`, `ethernet`, `wifi` etc). |
+| `connectivity.cellular.technology` | string | The type of a radio technology used for cellular connection |
+| `connectivity.cellular.carrier_name` | string | The name of the SIM carrier |
+
+### Global User Attributes
```
Comment (apigirl) [inline docs/mobile_data_collected.md:41]:
```suggestion
### Global user attributes
```

File: docs/mobile_data_collected.md [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+You can enable [tracking user info][2] globally to collect and apply user attributes to all RUM events.
+
+| Attribute name   | Type   | Description                 |
+|------------------|--------|-----------------------------|
+| `user.id`     | string | Identifier of the user. |
+| `usr.name` | string | Name of the user. |
+| `usr.email` | string | Email of the user. |
+
+
+## Event Specific Metrics and Attributes
```
Comment (apigirl) [inline docs/mobile_data_collected.md:52]:
```suggestion
## Event specific metrics and attributes
```

File: docs/mobile_data_collected.md [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+
+Metrics are quantifiable values that can be used for measurements related to the event. Attributes are non-quantifiable values used to slice metrics data (group by) in analytics. 
+
+{{% /tab %}}
+{{% tab "View" %}}
+
+### View Metrics
```
Comment (apigirl) [inline docs/mobile_data_collected.md:59]:
```suggestion
### View metrics
```

File: docs/mobile_data_collected.md [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+
+| `view.is_active`      |    boolean   | Indicates whether the View corresponding to this event is considered active                                                                                       
+      |
+
+
+
+{{% /tab %}}
+{{% tab "Resource" %}}
+
+### Resource Metrics
```
Comment (apigirl) [inline docs/mobile_data_collected.md:79]:
```suggestion
### Resource metrics
```

File: docs/mobile_data_collected.md [hunk 9]
```diff
@@ -0,0 +1,173 @@
+#Mobile Data Collected
+
+The Datadog Real User Monitoring SDK generates six types of events:
+
+| Event Type | Description                                                                                                                                                                                                                                                   |
```
Comment (ncreated) [inline docs/mobile_data_collected.md:5]: This table seems broken when I [preview its GH rendered page](https://github.com/DataDog/dd-sdk-ios/blob/priyanshi/data_collected/docs/mobile_data_collected.md).

File: docs/mobile_data_collected.md (hunk 1 above)
Comment (ncreated) [reply docs/mobile_data_collected.md:9]: Suggesting further change: `ViewController` → `UIViewController`
```suggestion
| View       | View represents a unique screen (or screen segment) on your mobile application. Individual `UIViewControllers` are classified as distinct views. While a user stays on a view, RUM event attributes (Errors, Resources, Actions) get attached to the view with a unique `view.id`                                   |
```

File: docs/mobile_data_collected.md [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+ |
+| View       | View represents a unique screen (or screen segment) on your mobile application. Individual ViewControllers are classified as distinct views. While a user stays on a view, RUM event attributes (Errors, Resources, Actions) get attached to the View with a unique `view.id`                                   |
+| Resource   | Resource represents network requests to first-party hosts, APIs, 3rd party providers and libraries in your mobile application. All requests generated during a user session are attached to the view with a unique `resource.id`                                                      |
+| Error     | Error represents an exception or crash (fatal error) emitted by the mobile application attached to the view it is generated in.                                                                                                                                            |
```
Comment (ncreated) [inline docs/mobile_data_collected.md:11]: `fatal error` may be misunderstood due to `fatalError()` Swift function causing a particular type of crash. I'd remove it at all and not specify what is a "crash" (every iOS dev knows that).
```suggestion
| Error     | Error represents an exception or crash emitted by the mobile application attached to the view it is generated in.                                                                                                                                            |
```

File: docs/mobile_data_collected.md [hunk 11]
```diff
@@ -0,0 +1,173 @@
...
+| Action   | Action represents user activity in your mobile application (application launch, tap, swipe, back etc). Each action is attached with a unique `action_id` attached to the view it gets generated in.                                                                                                                                                |
```
Comment (ncreated) [inline docs/mobile_data_collected.md:12]: For the sake of consistency with `view.id` and `resource.id` mentioned earlier:
```suggestion
| Action   | Action represents user activity in your mobile application (application launch, tap, swipe, back etc). Each action is attached with a unique `action.id` attached to the view it gets generated in.                                                                                                                                                |             
```

File: docs/mobile_data_collected.md [hunk 12]
```diff
@@ -0,0 +1,173 @@
...
+
+### Common Core Attributes
+
+| Attribute name   | Type   | Description                 |
+|------------------|--------|-----------------------------|
+| `type`     | string | The type of the event (for example, `view` or `resource`).             |
+| `application.id` | string | The Datadog application ID. |
+| `session.id` | string | Unique ID of the session. |
+| `view.id` | string | Unique ID of the initial view corresponding to the event. |
+| `view.url` | string | URL of the initial view corresponding to the event. |
```
Comment (ncreated) [inline docs/mobile_data_collected.md:31]: `view.url` is automatically computed from `UIViewController's` class name, so:
```suggestion
| `view.url` | string | Name of the `UIViewController` class corresponding to the event. |
```

File: docs/mobile_data_collected.md [hunk 13]
```diff
@@ -0,0 +1,173 @@
...
+| `view.name` | string | Name of the initial view corresponding to the event. |
```
Comment (ncreated) [inline docs/mobile_data_collected.md:32]: To make it more explanatory (`view.name` can be customized by the developer and it defaults to `view.url` if not):
```suggestion
| `view.name` | string | Custom name of the view corresponding to the event. |
```

File: docs/mobile_data_collected.md [hunk 14]
```diff
@@ -0,0 +1,173 @@
...
+| `service` | string | The [unified service name][] for this application used to corelate user sessions. |
```
Comment (ncreated) [inline docs/mobile_data_collected.md:33]: `[]` missing the link here, no?

File: docs/mobile_data_collected.md [hunk 15]
```diff
@@ -0,0 +1,173 @@
...
+| `date` | integer  | Start of the event in ms from epoch. |
+| `session.type` | string | Type of the session (`user`). |
+| `connectivity.status` | string | Status of device connectivity (`connected`, `not connected`, `maybe`). |
```
Comment (ncreated) [inline docs/mobile_data_collected.md:36]: "Network reachability" is a common term for iOS developers.
```suggestion
| `connectivity.status` | string | Status of device network reachability (`connected`, `not connected`, `maybe`). |
```

File: docs/mobile_data_collected.md [hunk 16]
```diff
@@ -0,0 +1,173 @@
...
+
+{{% /tab %}}
+{{% tab "View" %}}
+
+### View Metrics
+
+
+| Metric                              | Type        | Decription                                                                                          |
+|----------------------------------------|-------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `view.time_spent`                             | number (ns) | Time spent on the current view.                                                                                                                                                                                                  |
```
Comment (ncreated) [inline docs/mobile_data_collected.md:64]:
```suggestion
| `view.time_spent`                             | number (ns) | Time spent on the this view.                                                                                                                                                                                                  |
```

File: docs/mobile_data_collected.md [hunk 17]
```diff
@@ -0,0 +1,173 @@
...
+| `view.error.count`            | number      | Count of all errors collected for this view.                                                                                                                                                                |
+
+| `view.resource.count`         | number      | Count of all resources collected for this view.                                                                                                                                                                            |
+| `view.action.count`      | number      | Count of all actions collected for this view.                                                                                     
+      |
+
+| `view.is_active`      |    boolean   | Indicates whether the View corresponding to this event is considered active
```
Comment (ncreated) [inline docs/mobile_data_collected.md:71]: As we lowercase "view" everywhere else:
```suggestion
| `view.is_active`      |    boolean   | Indicates whether the view corresponding to this event is considered active                                                                                       
```

File: docs/mobile_data_collected.md [hunk 18]
```diff
@@ -0,0 +1,173 @@
...
+
+
+| Metric                              | Type           | Description                                                                                                                               |
+|----------------------------------------|----------------|-------------------------------------------------------------------------------------------------------------------------------------------|
+| `duration`                             | number         | Entire time spent loading the resource.                                                                                                   |
+| `resource.size`                | number (bytes) | Resource size.                                                                                                                            |
+| `resource.connect.duration`    | number (ns)    | Time spent establishing a connection to the server (connectEnd - connectStart)                                                            |
+| `resource.ssl.duration`        | number (ns)    | Time spent for the TLS handshake. If the last request is not over HTTPS, this metric does not appear (connectEnd - secureConnectionStart) |
```
Comment (ncreated) [inline docs/mobile_data_collected.md:87]: It's not true for the iOS SDK:
```suggestion
| `resource.ssl.duration`        | number (ns)    | Time spent for the TLS handshake. |
```

File: docs/mobile_data_collected.md [hunk 19]
```diff
@@ -0,0 +1,173 @@
...
+| `resource.dns.duration`        | number (ns)    | Time spent resolving the DNS name of the last request (domainLookupEnd - domainLookupStart)                                               |
+| `resource.redirect.duration`   | number (ns)    | Time spent on subsequent HTTP requests (redirectEnd - redirectStart)                                                                      |
+| `resource.first_byte.duration` | number (ns)    | Time spent waiting for the first byte of response to be received (responseStart - RequestStart)                                           |
```
Comment (ncreated) [inline docs/mobile_data_collected.md:90]:
```suggestion
| `resource.first_byte.duration` | number (ns)    | Time spent waiting for the first byte of response to be received (responseStart - requestStart)                                           |
```

File: docs/mobile_data_collected.md [hunk 20]
```diff
@@ -0,0 +1,173 @@
...
+| `resource.download.duration`   | number (ns)    | Time spent downloading the response (responseEnd - responseStart)                                                                         |
+
+### Resource attributes
+
+| Attribute                      | Type   | Description                                                                             |
+|--------------------------------|--------|-----------------------------------------------------------------------------------------|
+| `resource.id`                | string |  Unique identifier of the resource.      |
+| `resource.type`                | string | The type of resource being collected (for example, `xhr`, `image`, `font`, `CSS`, `Javascript`).          |
```
Comment (ncreated) [inline docs/mobile_data_collected.md:98]:
```suggestion
| `resource.type`                | string | The type of resource being collected (for example, `xhr`, `image`, `font`, `css`, `js`).          |
```

File: docs/mobile_data_collected.md [hunk 21]
```diff
@@ -0,0 +1,173 @@
...
+
+
+
+### Action metrics
+
+| Metric    | Type   | Description              |
+|--------------|--------|--------------------------|
+| `action.loading_time` | number (ns) | The loading time of the action.  |
+| `action.resource.count`         | number      | Count of all resources collected for this action. |
+| `action.error.count`      | number      | Count of all errors collected for this action.|
```
Comment (ncreated) [inline docs/mobile_data_collected.md:149]: "collected by" sounds technical. The user observation is IMO better expressed by "issued by" (as the action may result with resource start or error occurrence):
```suggestion
| `action.resource.count`         | number      | Count of all resources issued by this action. |
| `action.error.count`      | number      | Count of all errors issued by this action.|
```

**Review by ncreated — CHANGES_REQUESTED** [review]
Nice write up @priyanshi-gupta 💪. I left feedback, let's run next iteration 👍.

**Review by ncreated — APPROVED** [review]

**Pull request merged**